tests/core/full_node/stores/test_coin_store.py

```diff
@@ -498,3 +502,137 @@ async def test_unsupported_version() -> None:
     with pytest.raises(RuntimeError, match="CoinStore does not support database schema v1"):
         async with DBConnection(1) as db_wrapper:
             await CoinStore.create(db_wrapper)
+
+
+TEST_COIN_ID = b"c" * 32
+TEST_PUZZLEHASH = b"p" * 32
+TEST_AMOUNT = 1337
+TEST_PARENT_ID = Coin(b"a" * 32, TEST_PUZZLEHASH, TEST_AMOUNT).name()
+TEST_PARENT_PARENT_ID = b"f" * 32
+
+
+@dataclass(frozen=True)
+class UnspentLineageIdsTestItem:
+    coin_id: bytes
+    puzzlehash: bytes
+    amount: int
+    parent_id: bytes
+    is_spent: bool = False
+
+
+@dataclass
+class UnspentLineageIdsCase:
+    id: str
+    items: List[UnspentLineageIdsTestItem]
+    expected_success: bool
+    marks: Marks = ()
+
+
+@pytest.mark.anyio
+@datacases(
+    UnspentLineageIdsCase(
+        id="Unspent with parent that has same amount but different puzzlehash",
+        items=[
+            UnspentLineageIdsTestItem(TEST_COIN_ID, TEST_PUZZLEHASH, TEST_AMOUNT, TEST_PARENT_ID),
+            UnspentLineageIdsTestItem(b"2" * 32, b"2" * 32, 2, b"1" * 32),
+            UnspentLineageIdsTestItem(b"3" * 32, b"3" * 32, 3, b"2" * 32),
+            UnspentLineageIdsTestItem(TEST_PARENT_ID, b"4" * 32, TEST_AMOUNT, TEST_PARENT_PARENT_ID, is_spent=True),
+        ],
+        expected_success=False,
+    ),
+    UnspentLineageIdsCase(
+        id="Unspent with parent that has same puzzlehash but different amount",
```

Comment:
I suppose this test case is flipped too then

Reply:
Yep, I sent an adaptation as part of the second commit.